From fe50a3c8959f8c8e2abb9e18cb07f17385bc7f92 Mon Sep 17 00:00:00 2001
From: Sowmiya Sree Elavalagan <ssreeela@codeaurora.org>
Date: Wed, 24 Jun 2020 11:07:07 +0530
Subject: [PATCH] ath11k: disable qdss trace in ftm mode

EVM and DPD failures were seen in ftm mode as qdss trace was enabled
by default. Disabled qdss trace in ftm mode as it is not required.

Signed-off-by: Sowmiya Sree Elavalagan <ssreeela@codeaurora.org>
---
 drivers/net/wireless/ath/ath11k/qmi.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/wireless/ath/ath11k/qmi.c b/drivers/net/wireless/ath/ath11k/qmi.c
index 81651d2..e9d6d21 100644
--- a/drivers/net/wireless/ath/ath11k/qmi.c
+++ b/drivers/net/wireless/ath/ath11k/qmi.c
@@ -2465,6 +2465,10 @@ int ath11k_config_qdss(struct ath11k_base *ab)
 {
 	int ret;
 
+	/* Disabling qdss trace for FTM as it causes hig evt latency in FW */
+	if (ab->fw_mode == ATH11K_FIRMWARE_MODE_FTM)
+		return 0;
+
 	if (enable_qdss_trace) {
 		ret = ath11k_qmi_send_qdss_config(ab);
 		if (ret < 0) {
-- 
2.7.4

